pkg/operator/ceph/cluster/rbd/mirror.go

```diff
@@ -61,61 +62,59 @@ func (r *ReconcileCephRBDMirror) start(cephRBDMirror *cephv1.CephRBDMirror) erro
 
 	logger.Infof("configure rbd-mirroring with %d workers", cephRBDMirror.Spec.Count)
 
-	for i := 0; i < cephRBDMirror.Spec.Count; i++ {
-		daemonID := k8sutil.IndexToName(i)
-		resourceName := fmt.Sprintf("%s-%s", AppName, daemonID)
-		daemonConf := &daemonConfig{
-			DaemonID:     daemonID,
-			ResourceName: resourceName,
-			DataPathMap:  config.NewDatalessDaemonDataPathMap(cephRBDMirror.Namespace, r.cephClusterSpec.DataDirHostPath),
-			ownerRef:     *ref,
-		}
+	daemonID := k8sutil.IndexToName(0)
+	resourceName := fmt.Sprintf("%s-%s", AppName, daemonID)
+	daemonConf := &daemonConfig{
+		DaemonID:     daemonID,
```

Comment:
What if the `daemonID` were something like `mirroring` instead of `a`? Then it won't conflict with removal of the legacy daemon named `a`.